opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/export/__init__.py

```diff
@@ -128,7 +129,8 @@ def emit(self, log_data: LogData):
         try:
             self._exporter.export((log_data,))
         except Exception:  # pylint: disable=broad-exception-caught
-            _logger.exception("Exception while exporting logs.")
+            traceback_str = traceback.format_exc()
+            print(f"Exception while exporting logs: {traceback_str}")
```

Comment:
the print to stdout can potentially have unwanted side-effects, e.g. when an application communicates with other processes via pipes.

additionally, i think the print doesn't fully solve the original recursive logging problem since an exporter might still emit its own log messages.

since the `SimpleLogRecordProcessor` is already setting the [suppress instrumentation](https://github.com/open-telemetry/opentelemetry-python/pull/4088/files#diff-f72d80bed91a7ebff9f47b509561d42e57edacd454317801f445d6ba4226d4c8R128) key on the context, would it maybe make sense to consider this in the `LoggingHandler`/`Logger` and skip emitting the logs accordingly?